webapp/src/components/AssessmentsPage.tsx

```diff
@@ -0,0 +1,410 @@
+                    ]}
+                  >
+                    <span style={{ fontWeight: 'bold' }}>
+                      {assessment.title}
+                    </span>
+                    <br />
+                    {assessment.description}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[
+                      StatusTab.All,
+                      StatusTab.Active,
+                      StatusTab.Past,
+                      StatusTab.Upcoming,
+                    ]}
+                  >
+                    {assessment.type}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[StatusTab.All, StatusTab.Active]}
+                  >
+                    {formatDateTime(assessment.dueDate)}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[StatusTab.Active]}
+                  >
+                    {assessment.type === 'Test' && assessment.testDuration}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[StatusTab.Past]}
+                  >
+                    {(assessment.status === 'Submitted' ||
+                      assessment.status === 'Graded') &&
+                      formatDateTime(assessment.submittedDate)}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[StatusTab.All, StatusTab.Past]}
+                  >
+                    {assessment.score !== -1 && assessment.score}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[StatusTab.Upcoming]}
+                  >
+                    {formatDateTime(assessment.availableDate)}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[
+                      StatusTab.All,
+                      StatusTab.Active,
+                      StatusTab.Past,
+                      StatusTab.Upcoming,
+                    ]}
+                  >
+                    {renderChipByStatus(assessment.status)}
+                  </TableCellWrapper>
+                  <TableCellWrapper
+                    statusTab={currentStatusTab}
+                    index={[StatusTab.All, StatusTab.Active, StatusTab.Past]}
+                  >
+                    {renderButtonByStatus(assessment.status)}
+                  </TableCellWrapper>
+                </TableRowWrapper>
+              ))
+            ) : (
+              <div style={{ padding: '20px' }}>
```

Comment:
It would be invalid markup to have a `<div>` inside of a table body, but of even more concern: we should be hiding the tabs and the table altogether if there are no assessments, not just having an empty table.